# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/ScriptLoader.cpp
# Commit: f3204b9ec0dc
# Full Hash: f3204b9ec0dc99aad004ce80499052fe5dcd5494
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2023-01-19 04:46:07
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement requestCredentials for Module Workers; r=jonco
#   
#   This weaves credentials through from the private to script creation.
#   
#   Depends on D155567
# ==============================================================================

diff -r 7327e6438343 -r f3204b9ec0dc dom/workers/ScriptLoader.cpp
--- a/dom/workers/ScriptLoader.cpp	Wed Jan 18 13:46:31 2023 +0000
+++ b/dom/workers/ScriptLoader.cpp	Wed Jan 18 13:46:31 2023 +0000
@@ -529,7 +529,7 @@
     loadContext->mLoadResult = rv;
   }
 
-  RefPtr<ScriptLoadRequest> request;
+  RefPtr<ScriptLoadRequest> request = nullptr;
   if (mWorkerRef->Private()->WorkerType() == WorkerType::Classic) {
     RefPtr<ScriptFetchOptions> fetchOptions =
         new ScriptFetchOptions(CORSMode::CORS_NONE, referrerPolicy, nullptr);
@@ -539,12 +539,19 @@
                                     loadContext);
   } else {
     // Implements part of "To fetch a worklet/module worker script graph"
-    // including, setting up the request with a credentials mode (TODO),
+    // including, setting up the request with a credentials mode,
     // destination (CSP, TODO).
 
-    // Step 1. Let options be a script fetch options. TODO: credentials mode.
+    // Step 1. Let options be a script fetch options.
+    // We currently don't track credentials in our ScriptFetchOptions
+    // implementation. We translate this instead to cors.
+    CORSMode cors = mWorkerRef->Private()->WorkerCredentials() ==
+                            RequestCredentials::Include
+                        ? CORSMode::CORS_USE_CREDENTIALS
+                        : CORSMode::CORS_ANONYMOUS;
+
     RefPtr<ScriptFetchOptions> fetchOptions =
-        new ScriptFetchOptions(CORSMode::CORS_NONE, referrerPolicy, nullptr);
+        new ScriptFetchOptions(cors, referrerPolicy, nullptr);
 
     RefPtr<WorkerModuleLoader::ModuleLoaderBase> moduleLoader =
         GetGlobal()->GetModuleLoader(nullptr);